src/mainwindow.cpp

```diff
@@ -2340,6 +2340,36 @@ void MainWindow::resizeOptionEditor(const QSize &size)
     this->resizeDocks({ui->dockOptionEditor}, {size.height()}, Qt::Vertical);
 }
 
+
+void MainWindow::setForeground()
```

Comment:
@pandahs I just moved the else part to this function... What will happen if some other developer used your original function?